### DISCUSSION TITLE
RFC: Workflow and subject set selection

### DISCUSSION THREAD

**Discussion by beckyrother** [zooniverse/front-end-monorepo#2445] [General]
# Workflow selection
Possible paths to Classify

## From Project Home Page ([InVision](https://invis.io/9WNF9M73AEN#/432001828_1_Choose_Workflow))

**1. Single workflow, no subject set selection**
    - Click a **gold** button on project home page to start with a random subject ![image](https://user-images.githubusercontent.com/1392130/106055970-b5c60c80-60b3-11eb-9e98-c05cad816a00.png)

**2. Single workflow, subject set selection**
    - Click a **teal** color button with a `set selection` label to see a modal on top of the project home page allowing for subject set selection ![Zooniverse_Screens_-_InVision](https://user-images.githubusercontent.com/1392130/106056928-d04cb580-60b4-11eb-8c55-c3778d43a7ae.png)

**3. Single workflow, subject set selection, subject selection with sequential subject ordering**
    - Same as above, plus:
    - Second screen within modal allowing for subject selection. Messaging within modal indicates that subjects will be sequential. 

**4. Multiple workflows**
    - Click the button of the desired workflow to follow one of the above paths. Ideally, workflow options should be divided by workflow selection or not: 
    ![image](https://user-images.githubusercontent.com/1392130/106058537-fd9a6300-60b6-11eb-8bcd-3c6374e7f95a.png)

---

## From Talk, etc

Clicking Classify in the nav bar will always navigate to a classify interface; however, depending on project setup different things will happen:

**1. Single workflow, no subject set selection**
   - Classify random subject
 
**2. Single workflow, subject set selection**
    - Go to classify interface, modal opens (the same one as above) to allow for subject set selection

**3. Single workflow, subject set selection, subject selection with sequential subject ordering**
    - Go to classify interface, modal opens (same as above)

**4. Multiple workflows**
    - Go to classify interface, modal opens with workflow selection -> subject set selection -> subject selection (path ends as specified by project)
    - 
![image](https://user-images.githubusercontent.com/1392130/106060090-405d3a80-60b9-11eb-8c4c-f092eaf5d630.png)

---

## From a bookmarked Classify page

**1. If workflow, subject set, etc are still active**
    - Go directly to classifying subject, or open a modal as described above 

**2. If part of the path is retired**
    - Redirect to project home page with small popup explaining that the classify page is no longer active and asking to choose a different workflow, etc. [InVision](https://invis.io/9WNF9M73AEN#/443437830_1_Choose_Workflow_Copy)
    - 
![image](https://user-images.githubusercontent.com/1392130/106061129-91216300-60ba-11eb-96e3-a4aa7ae78624.png)

**Comment by eatyourgreens** [comment]
As a project owner, if you've selected the `grouped` flag for your workflow then we may have to show you the subject set selector regardless. The API request for subjects will fail without a `subject_set` parameter. We could automatically set that for you but I'm not sure if we should spend much time supporting edge cases. Grouped selection doesn't make sense without a choice of subject sets. Maybe the project builder should prevent you from selecting the `grouped` option unless your workflow satisfies some requirements, like having more than one subject set?

Will the most common case be a transcription-type project where there's a single workflow, but many subject sets, each representing a volume/file that needs to be transcribed?

**Comment by srallen** [comment]
How do we support a project that has a single workflow, single subject set, but wants you to select a subject? Basically the case that we're eventually supposed to support for CSSI.

**Comment by eatyourgreens** [comment]
That's a good question. We have a prototype that allows searching a smallish subject set for subjects (#1925) but searching or browsing all subjects linked to a workflow isn't supported.

**Comment by eatyourgreens** [comment]
In #1875 I've added `/classify/workflow/[workflowID]/subject/[subjectID]` as a possible URL for specific subjects.

**Comment by beckyrother** [comment]
Any workflow that allows for choice (basically anything that opens a modal rather than going directly to the classify page) should be teal. I think `grouped` falls under that category, as does the transcription-type project with only one workflow but multiple subject sets
